File: 09_Network/swiperefreshlayout/src/main/java/com/codingnight/example/swiperefreshlayout/MainActivity.java
package com.codingnight.example.swiperefreshlayout;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.DataSource;
import com.bumptech.glide.load.engine.GlideException;
import com.bumptech.glide.request.RequestListener;
import com.bumptech.glide.request.target.Target;

import java.util.Random;

public class MainActivity extends AppCompatActivity {

    ImageView imageView;
    SwipeRefreshLayout swipeRefreshLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        imageView = findViewById(R.id.imageView);
        swipeRefreshLayout = findViewById(R.id.swipeRefreshLayout);

        swipeRefreshLayout.setOnRefreshListener(this::imageLoaderWithLruCache);

        imageLoaderWithLruCache();
    }

    private void imageLoaderWithLruCache() {
        String url1 = "https://cdn.pixabay.com/photo/2016/01/08/05/24/sunflower-1127174_150.jpg";
        String url2 = "https://cdn.pixabay.com/photo/2021/08/15/06/54/sunflower-6546993_150.jpg";
        String url = new Random().nextBoolean() ? url1 : url2;
        Glide.with(this)
                .load(url)
                .placeholder(R.drawable.ic_launcher_background)
                .listener(new RequestListener<Drawable>() {
                    @Override
                    public boolean onLoadFailed(@Nullable GlideException e, Object model, Target<Drawable> target, boolean isFirstResource) {
                        if (swipeRefreshLayout.isRefreshing()) {
                            swipeRefreshLayout.setRefreshing(false);
                        }
                        return false;
                    }

                    @Override
                    public boolean onResourceReady(Drawable resource, Object model, Target<Drawable> target, DataSource dataSource, boolean isFirstResource) {
                        if (swipeRefreshLayout.isRefreshing()) {
                            swipeRefreshLayout.setRefreshing(false);
                        }
                        return false;
                    }
                })
                .into(imageView);
    }

}

File: 05_Storage/01_Sharedpreference/app/src/main/java/com/codeingnight/android/sharedpreference/MainActivity.kt
package com.codeingnight.android.sharedpreference

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        var shp = getPreferences(Context.MODE_PRIVATE)

//        var shp = getSharedPreferences(getString(R.string.file_key), MODE_PRIVATE)
//        with (shp.edit()) {
//            putInt("NUMBER", 100)
//            apply()
//        }
//        var x = getSharedPreferences(getString(R.string.file_key), MODE_PRIVATE).getInt("NUMBER", 0)
//        var TAG = "MyLog"
//        Log.d(TAG, "onCreate: " + x)

        val myData = MyData(applicationContext)
        myData.save()
        myData.load()
    }
}